chain-test-utils/src/time/mod.rs

```diff
@@ -0,0 +1,136 @@
+use chain_time::{Epoch, Slot, TimeEra};
+use smoke::{Generator, R};
+
+// Generate an `Epoch` from a generator function
+fn generate_epoch_with<GenF>(mut gen: GenF) -> Epoch
+where
+    GenF: FnMut() -> u32,
+{
+    Epoch(gen())
+}
+
+// Generate an `Slot` from a generator function
+fn generate_slot_with<GenF>(mut gen: GenF) -> Slot
+where
+    GenF: FnMut() -> u64,
+{
+    gen().into()
+}
+
+// Generate an `Epoch` given a `smoke::R` (random generator)
+pub fn generate_epoch(r: &mut R) -> Epoch {
+    generate_epoch_with(|| r.num())
+}
+
+// Generate an `Epoch` given a `smoke::R` (random generator) and a `(u32, u32)` range limit tuple
+pub fn generate_epoch_with_range(r: &mut R, range: (u32, u32)) -> Epoch {
+    generate_epoch_with(|| r.num_range(range.0, range.1))
+}
+
+// Generate an `Slot` given a `smoke::R` (random generator)
+pub fn generate_slot(r: &mut R) -> Slot {
+    generate_slot_with(|| r.num())
+}
```

Comment:
Can this generate nonsensical slot values?
I'm trying to understand the purpose of non-bounded random generator methods here.

Reply:
Yes, it actually just makes sense if you want to test both acceptable values and not.